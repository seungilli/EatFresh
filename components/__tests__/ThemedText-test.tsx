import * as React from "react";
import renderer from "react-test-renderer";
import { ThemedText } from "../ThemedText";

describe("<ThemedText />", () => {
  it("should render with default styles", () => {
    const tree = renderer
      .create(<ThemedText>Default text</ThemedText>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render with title styles", () => {
    const tree = renderer
      .create(<ThemedText type="title">Title text</ThemedText>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render with subtitle styles", () => {
    const tree = renderer
      .create(<ThemedText type="subtitle">Subtitle text</ThemedText>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render with defaultSemiBold styles", () => {
    const tree = renderer
      .create(<ThemedText type="defaultSemiBold">Semi-bold text</ThemedText>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render with link styles", () => {
    const tree = renderer
      .create(<ThemedText type="link">Link text</ThemedText>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should apply light color", () => {
    const tree = renderer
      .create(
        <ThemedText lightColor="#ffffff" darkColor="#000000">
          Light color text
        </ThemedText>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should apply dark color", () => {
    const tree = renderer
      .create(
        <ThemedText lightColor="#ffffff" darkColor="#000000">
          Dark color text
        </ThemedText>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should merge custom styles", () => {
    const tree = renderer
      .create(
        <ThemedText style={{ fontSize: 18, textAlign: "center" }}>
          Custom styled text
        </ThemedText>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render without crashing with empty props", () => {
    const tree = renderer.create(<ThemedText />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
