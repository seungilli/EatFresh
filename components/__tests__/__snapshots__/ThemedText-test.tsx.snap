// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedText component should apply dark color 1`] = `
<Text
  style={
    [
      {
        "color": "#ffffff",
      },
      {
        "fontSize": 16,
        "lineHeight": 24,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Dark color text
</Text>
`;

exports[`ThemedText component should apply light color 1`] = `
<Text
  style={
    [
      {
        "color": "#ffffff",
      },
      {
        "fontSize": 16,
        "lineHeight": 24,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Light color text
</Text>
`;

exports[`ThemedText component should merge custom styles 1`] = `
<Text
  style={
    [
      {
        "color": "#000000",
      },
      {
        "fontSize": 16,
        "lineHeight": 24,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      {
        "fontSize": 18,
        "textAlign": "center",
      },
    ]
  }
>
  Custom styled text
</Text>
`;

exports[`ThemedText component should render with default styles 1`] = `
<Text
  style={
    [
      {
        "color": "#000000",
      },
      {
        "fontSize": 16,
        "lineHeight": 24,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Default text
</Text>
`;

exports[`ThemedText component should render with defaultSemiBold styles 1`] = `
<Text
  style={
    [
      {
        "color": "#000000",
      },
      undefined,
      undefined,
      {
        "fontSize": 16,
        "fontWeight": "600",
        "lineHeight": 24,
      },
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Semi-bold text
</Text>
`;

exports[`ThemedText component should render with link styles 1`] = `
<Text
  style={
    [
      {
        "color": "#000000",
      },
      undefined,
      undefined,
      undefined,
      undefined,
      {
        "color": "#0a7ea4",
        "fontSize": 16,
        "lineHeight": 30,
      },
      undefined,
    ]
  }
>
  Link text
</Text>
`;

exports[`ThemedText component should render with subtitle styles 1`] = `
<Text
  style={
    [
      {
        "color": "#000000",
      },
      undefined,
      undefined,
      undefined,
      {
        "fontSize": 20,
        "fontWeight": "bold",
      },
      undefined,
      undefined,
    ]
  }
>
  Subtitle text
</Text>
`;

exports[`ThemedText component should render with title styles 1`] = `
<Text
  style={
    [
      {
        "color": "#000000",
      },
      undefined,
      {
        "fontSize": 32,
        "fontWeight": "bold",
        "lineHeight": 32,
      },
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Title text
</Text>
`;

exports[`ThemedText component should render without crashing with empty props 1`] = `
<Text
  style={
    [
      {
        "color": "#000000",
      },
      {
        "fontSize": 16,
        "lineHeight": 24,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
/>
`;
